SaraAlert/SaraAlert
# SARAALERT-1280: Sort Jurisdiction List Alphabetically

# Description
Jira Ticket: [SARAALERT-1280](https://tracker.codev.mitre.org/browse/SARAALERT-1280)

Sorts the jurisdiction list alphabetically when displayed in the UI.

## Feature - Screenshot
![image](https://user-images.githubusercontent.com/42656458/114775823-bc256580-9d3f-11eb-9321-4ee43e7252df.png)

## Important Changes

`app/javascript/components/admin/UserModal.js`
`app/javascript/components/enrollment/steps/Exposure.js`
`app/javascript/components/patient/monitoring_actions/Jurisdiction.js`
`app/javascript/components/public_health/query/JurisdictionFilter.js`
- New state variable holding the jurisdiction list which is also sorted alphabetically

## Recommended Steps for Testing

1. Add the following to the end of your local `jurisdiction.yml` file:
```
        'X Town':
        'B Town':
        'A Town':
            children:
                'Region BC':
                'Region AB':
                'Region':
        '123 Town':
```
Here is an example screenshot of how this addition should look in the file when you are done:
![image](https://user-images.githubusercontent.com/42656458/114776377-6ac9a600-9d40-11eb-9a5f-606d82c1e497.png)

2. Run the following command which will configure your local database with these additional jurisdictions: `bundle exec rake admin:import_or_update_jurisdictions`
3. Verify in Sara Alert that the new jurisdictions are now visible in the jurisdiction drop downs when you are logged in as a USA level user.
3. In each of the following locations, verify the jurisdiction list that appears is sorted alphabetically 
  - New Sara Alert user modal on the Admin dashboard
  - Jurisdiction Filter on the dashboard & custom export
  - Assigned Jurisdiction field in the Monitoree's Profile in the Monitoring Actions panel
  - Assigned Jurisdiction field in the Enrollment Potential Exposure Information page
5. Verify that selecting a jurisdiction from the list behaves as expected in each respective location.

# Checklists

**Submitter:**
- [x] **N/A** If UI changes have been made, Chrome Dev Tools Lighthouse accessibility test has been executed to ensure no 508 issues were introduced.
- [x] **N/A** Tests are included and test edge cases
- [x] **N/A** Test fixtures updated and documented as necessary


@DPC15038 :


@timwongj  :


@holmesie :
